### PR TITLE
fix: trader plugin CMC/Gecko token tag display glitch

### LINKED ISSUE #3024: [Bug] Trader plugin CMC/Gecko token tag display glitch
## Bug Info

Screen Shot 2021-04-26 at 17 21 09
Screen Shot 2021-04-26 at 17 23 05
Screen Shot 2021-04-26 at 17 27 57

Should add `space` or group the tags like CMC page :

Screen Shot 2021-04-26 at 17 29 54

### How to reproduce?

1.
2.
3.

### Mask Version:

- [x] (Browser) Installed from the extension store (version: )
- [ ] (App) Installed from the app store (version: )
- [ ] master branch
- [ ] Build of version/commit hash:

(Please write down build command/flags if there is anything special)

### System

 
- [ ] Windows 10
- [x] Mac OS X Catalina
- [ ] Linux: 

### Platform/Browser

 
- [x] Latest Chrome (stable channel)
- [x] Latest Firefox (stable channel)
- [ ] Android 11
- [ ] iOS 14

### PR DESCRIPTION
**Pull request by lelenei**
closes #3024

### CODE REVIEW

**Comment by lelenei** [comment]
image

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTable.tsx [hunk 1]
```diff
@@ -80,6 +78,10 @@ export function CoinMetadataTable(props: CoinMetadataTableProps) {
         ['Community', trending.coin.community_urls],
     ] as [string, string[] | undefined][]
 
+    console.log('-------')
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTable.tsx:81]: Remove it.

**Comment by guanbinrui** [comment]
image

@lelenei The last line is not aligned.

**Comment by lelenei** [comment]
image

**Comment by guanbinrui** [comment]
Please use [Chip](https://next.material-ui.com/components/chips/).

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    onClose: () => void
+}
+
+function TagsDialog(props: TagsDialogProps) {
+    const { open, tags, onClose } = props
+    const classes = useStyles()
+    return (
+        <InjectedDialog open={open} title="Tags" onClose={onClose}>
+            <DialogContent className={classes.content}>
+                {tags?.map((x, i) => (
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx:79]: if tags are empty. Nothing will render.

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+export interface CoinMetadataTagsProps {
+    tags?: string[]
+}
+
+export function CoinMetadataTags(props: CoinMetadataTagsProps) {
+    const classes = useStyles()
+    const { tags } = props
+    const [openSeaAll, setOpenSeaAll] = useState(false)
+
+    const OpenSeaAllDialog = useCallback(() => {
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx:39]: Use a proper name.

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+export interface CoinMetadataTagsProps {
+    tags?: string[]
+}
+
+export function CoinMetadataTags(props: CoinMetadataTagsProps) {
+    const classes = useStyles()
+    const { tags } = props
+    const [open, setOpen] = useState(false)
+
+    const OpenDialog = useCallback(() => {
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx:26]: Please use a proper name.

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        setOpen(!open)
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx:27]: `open` not in the deps list.

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTable.tsx [hunk 6]
```diff
@@ -46,11 +47,9 @@ const useStyles = makeStyles((theme) =>
                 paddingRight: 0,
             },
         },
-        tag: {
-            paddingRight: theme.spacing(2),
-            '&:last-child': {
-                paddingRight: 0,
-            },
+        tagCell: {
+            display: 'flex',
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTable.tsx:51]: Suppose you haven't changed the `display` of the table cell. It's unnecessary to set the margin-bottom property.

**Review by septs — DISMISSED** [review]

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+export function CoinMetadataTags(props: CoinMetadataTagsProps) {
+    const classes = useStyles()
+    const { tags } = props
+    const [open, setOpen] = useState(false)
+
+    const onClick = useCallback(() => {
+        setOpen(!open)
+    }, [open, setOpen])
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx:28]:
```suggestion
    const onClick = useCallback(() => {
        setOpen((open) => !open)
    }, [])
```

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+    const onClose = useCallback(() => {
+        setOpen(false)
+    }, [setOpen])
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx:32]:
```suggestion
    const onClose = useCallback(() => {
        setOpen(false)
    }, [])
```

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    return (
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx:33]:
```suggestion
    if (!tags) return null
    return (
```

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        <>
+            {tags?.map((x, i) =>
+                i < 4 ? (
+                    <Linking key={i} href={x} LinkProps={{ className: classes.tag }}>
+                        <Chip label={x.replace(/-/g, ' ')} size="small" />
+                    </Linking>
+                ) : null,
+            )}
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx:41]:
```suggestion
            {tags.slice(0, 4).map((x, i) =>
                <Linking key={i} href={x} LinkProps={{ className: classes.tag }}>
                    <Chip label={x.replace(/-/g, ' ')} size="small" />
                </Linking>
            )}
```

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+            {tags?.length! > 4 ? (
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx:42]:
```suggestion
            {tags.length > 4 ? (
```

File: packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+                <>
+                    <Linking key={tags?.length! + 1} href={'View all'} LinkProps={{ className: classes.tag }}>
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Trader/UI/trending/CoinMetadataTags.tsx:44]:
```suggestion
                    <Linking key={tags.length + 1} href={'View all'} LinkProps={{ className: classes.tag }}>
```

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**